tsdb/engine/tsm1/engine.go

```diff
@@ -1947,9 +1937,13 @@ func (e *Engine) WriteSnapshot() (err error) {
 }
 
 // CreateSnapshot will create a temp directory that holds
-// temporary hardlinks to the underylyng shard files.
-func (e *Engine) CreateSnapshot() (string, error) {
-	if err := e.WriteSnapshot(); err != nil {
+// temporary hardlinks to the underlying shard files.
+// skipCacheOk controls whether it is permissible to fail writing out
+// in-memory cache data when a previous snapshot is in progress
+func (e *Engine) CreateSnapshot(skipCacheOk bool) (string, error) {
+	if err := e.WriteSnapshot(); (err == ErrSnapshotInProgress) && skipCacheOk {
+		e.logger.Warn("Snapshotter busy: Backup or export proceeding without cache snapshot contents.")
```

Comment:
Probably should change this log message to something like `cache snapshotter busy: proceeding without cache contents` since this function can potentially be called for other reasons. I.e., this function doesn't know why it's being called.

Reply:
Changed, will push.

Reply:
Unit test added in https://github.com/influxdata/influxdb/pull/19869/commits/0dcff81f562809ab471958bb68b4557e4f6aced2

Reply:
Comment changed in https://github.com/influxdata/influxdb/pull/19869/commits/6ec446f4222e60f07c698a973cd13a45553c0834